src/types/impl-trait.md

```diff
@@ -5,33 +5,120 @@
 >
 > _ImplTraitTypeOneBound_ : `impl` [_TraitBound_]
 
-## Anonymous type parameters
+> **Edition differences**: `impl Trait` is new in the 2018 edition.
```

Comment:
```suggestion
```

The edition differences are only for things that change compilation based on the edition.  The old style of the edition guide presenting all new features as part of the "edition" is being phased out.

Reply:
Thanks for the review! I had seen other parts of the reference using language like this. Is text like "`impl Trait` is not available in the 2015 edition" better? I think I saw some style guide saying to prefer that, but I can't seem to find it now.

Reply:
impl trait can be used on both 2015 and 2018.  The only differences between the editions are listed here: https://doc.rust-lang.org/edition-guide/rust-2018/edition-changes.html

The reference conventions are listed [here](https://doc.rust-lang.org/nightly/reference/#conventions) and there is a style guide [here](https://github.com/rust-lang/reference/blob/master/STYLE.md).

Reply:
Oh, I see. Apparently some new features were enabled in the 2015 edition after 2015, and only some (with syntax backward-incompatibilities?) were gated on edition=2018? I'm fine with deleting that note, then. (Though I also think the edition guide could use some better explanation about how new features found their way into the 2015 edition.)

Reply:
Yea. Usually new features work on all editions if possible. Only if it is backwards-incompatible in some way does it get gated on a new edition.  FWIW, the edition guide is in the process of being rewritten.